keyboards/alps64/keymaps/numist/keymap.c

```diff
@@ -0,0 +1,124 @@
+#include QMK_KEYBOARD_H
+
+enum custom_keycodes {
+    M_FIND = SAFE_RANGE,
+    M_AGAIN,
+    M_UNDO,
+    M_CUT,
+    M_COPY,
+    M_PSTE,
+    DYNAMIC_MACRO_RANGE
+};
+
+#define DYNAMIC_MACRO_SIZE 64
+#define DYNAMIC_MACRO_RANGE SAFE_RANGE
+#include "dynamic_macro.h"
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    bool use_cmd = true;
+    if (keymap_config.swap_lalt_lgui == 1) {
+        use_cmd = false;
+    }
+
+    switch (keycode) {
+        case M_UNDO:
```

Comment:
This entire block could be removed, actually. 

If you use `LCTL(KC_Z)` in your keymap, "bootmagic" will take care of this.  Basically, support for all of this is already built into QMK Firmware. 

And, you'd want to use `CG_*` instead of the `AG_*` keycodes:
https://docs.qmk.fm/#/feature_bootmagic?id=keycodes